e2e/cluster.go

```diff
@@ -366,6 +375,26 @@ func prepareClusterForTests(
 	return kube, subnetId, clusterParams, nil
 }
 
+// TODO(cameissner): figure out a better way to reconcile server-side and client-side properties,
+// for now we simply regenerate a new base model and manually patch its properties according to the original model
+func prepareClusterModelForRecreate(r *mrand.Rand, clusterModel *armcontainerservice.ManagedCluster) (*armcontainerservice.ManagedCluster, error) {
+	if clusterModel == nil || clusterModel.Properties == nil {
+		return nil, fmt.Errorf("unable to prepare cluster model for recreate, got nil cluster model/properties")
```

Comment:
okay a little silly but fine for now

Reply:
yep, will make a point of coming back to this